### DISCUSSION TITLE
Compile ESP32 Arduino code on ESP32-S3

### DISCUSSION THREAD

**Discussion by RodEnry** [espressif/arduino-esp32#7576] [General]
Hello, 
I have a code written for ESP32-WROOM-32 that includes several libraries (for example, accelerometer, gyroscope, Webserver, Sd card etc...). 

Is it possible to compile the same code for the ESP32-S3-DevKitC?

Thanks
Enry

**Comment by me-no-dev** [comment]
Yes, absolutely. Just make sure that you select the correct GPIOs

**Reply by RodEnry** [reply]
Many thanks for your fast reply!
That's a very nice news. So basically I can keep the whole code base and upgrade to the new hardware!
I'll check and change some GPIOs since some of them are different.

About the firmware update, is it possible to update the firmware straight via USB? I mean, the USB bootloader should be native in the chip right, also after erasing the whole flash... is that right?

Enry

**Reply by me-no-dev** [reply]
Yes, you can use the built-in USB port to flash/erase/communicate with the chip
